File: refactor.js
const fs = require('fs');
const path = require('path');

const tabs = [
  'MaintenanceTab.tsx',
  'InventoryTab.tsx',
  'PurchasesTab.tsx',
  'InvoicingTab.tsx',
  'ReportsTab.tsx',
  'ReportsViewTab.tsx',
  'RolesTab.tsx',
  'PersonalTab.tsx',
  'ConfigTab.tsx',
  'UsersTab.tsx',
  'SuperAdminTab.tsx'
];

const dir = path.join(__dirname, 'src', 'components');

const replacements = [
  [/\bbg-white\b/g, 'bg-[var(--modal-bg)]'],
  [/\bborder-zinc-100\b/g, 'border-[var(--border-main)]'],
  [/\bborder-zinc-200\b/g, 'border-[var(--border-main)]'],
  [/\bborder-zinc-300\b/g, 'border-[var(--border-main)]'],
  [/\btext-zinc-900\b/g, 'text-[var(--text-main)]'],
  [/\btext-zinc-800\b/g, 'text-[var(--text-main)]'],
  [/\btext-zinc-500\b/g, 'text-[var(--text-muted)]'],
  [/\btext-zinc-400\b/g, 'text-[var(--text-muted)]'],
  [/\btext-gray-500\b/g, 'text-[var(--text-muted)]'],
  [/\btext-gray-400\b/g, 'text-[var(--text-muted)]'],
  [/\btext-slate-800\b/g, 'text-[var(--text-main)]'],
  [/\btext-slate-500\b/g, 'text-[var(--text-muted)]'],
  [/\bhover:bg-zinc-50\b/g, 'hover:bg-[var(--table-row-hover)]'],
  [/\bhover:bg-zinc-100\b/g, 'hover:bg-[var(--table-row-hover)]'],
  [/\bbg-zinc-100\b/g, 'bg-[var(--table-header-bg)]'],
  [/\bbg-zinc-50\b/g, 'bg-[var(--table-header-bg)]'],
  [/\bdivide-zinc-200\b/g, 'divide-[var(--table-divider)]'],
  [/\bdivide-zinc-100\b/g, 'divide-[var(--table-divider)]'],
  [/\bborder-slate-200\b/g, 'border-[var(--input-border)]'],
  [/\bborder-gray-100\b/g, 'border-[var(--border-main)]'],
  [/\bborder-gray-200\b/g, 'border-[var(--border-main)]'],
  [/\bdivide-slate-200\b/g, 'divide-[var(--table-divider)]']
];

tabs.forEach(tab => {
  const filepath = path.join(dir, tab);
  if (!fs.existsSync(filepath)) {
    console.error(`File not found: ${filepath}`);
    return;
  }
  
  let content = fs.readFileSync(filepath, 'utf8');

  replacements.forEach(([regex, repl]) => {
    content = content.replace(regex, repl);
  });

  fs.writeFileSync(filepath, content);
  console.log(`Updated ${tab}`);
});
console.log('done');

File: refactor.py
import os

tabs = [
  'InventoryTab.tsx',
  'PurchasesTab.tsx',
  'InvoicingTab.tsx',
  'ReportsTab.tsx',
  'ReportsViewTab.tsx',
  'RolesTab.tsx',
  'PersonalTab.tsx',
  'ConfigTab.tsx',
  'UsersTab.tsx',
  'SuperAdminTab.tsx',
  'StaffLogin.tsx',
  'Login.tsx',
  'SettingsModal.tsx'
]

dir_path = r'e:\VISUAL STUDIO CODE\MEKA TALLER\src\components'

replacements = {
  'bg-white': 'bg-[var(--modal-bg)]',
  'border-zinc-100': 'border-[var(--border-main)]',
  'border-zinc-200': 'border-[var(--border-main)]',
  'border-zinc-300': 'border-[var(--border-main)]',
  'text-zinc-900': 'text-[var(--text-main)]',
  'text-zinc-800': 'text-[var(--text-main)]',
  'text-zinc-500': 'text-[var(--text-muted)]',
  'text-zinc-400': 'text-[var(--text-muted)]',
  'text-gray-500': 'text-[var(--text-muted)]',
  'text-gray-400': 'text-[var(--text-muted)]',
  'text-slate-800': 'text-[var(--text-main)]',
  'text-slate-500': 'text-[var(--text-muted)]',
  'hover:bg-zinc-50': 'hover:bg-[var(--table-row-hover)]',
  'hover:bg-zinc-100': 'hover:bg-[var(--table-row-hover)]',
  'bg-zinc-100': 'bg-[var(--table-header-bg)]',
  'bg-zinc-50': 'bg-[var(--table-header-bg)]',
  'divide-zinc-200': 'divide-[var(--table-divider)]',
  'divide-zinc-100': 'divide-[var(--table-divider)]',
  'border-slate-200': 'border-[var(--input-border)]',
  'border-gray-100': 'border-[var(--border-main)]',
  'border-gray-200': 'border-[var(--border-main)]',
  'divide-slate-200': 'divide-[var(--table-divider)]'
}

for tab in tabs:
    filepath = os.path.join(dir_path, tab)
    if not os.path.exists(filepath):
        continue
        
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
        
    for old, new in replacements.items():
        content = content.replace(old, new)
        
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(content)
        
print("done")
